plugins/events-node/src/extensions.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createExtensionPoint } from '@backstage/backend-plugin-api';
+import {
+  EventBroker,
+  EventPublisher,
+  EventSubscriber,
+  RequestValidator,
+} from './api';
+
+/**
+ * @alpha
+ */
+export interface EventsExtensionPoint {
+  setEventBroker(eventBroker: EventBroker): void;
+
+  addPublishers(
+    ...publishers: Array<EventPublisher | Array<EventPublisher>>
+  ): void;
+
+  addSubscribers(
+    ...subscribers: Array<EventSubscriber | Array<EventSubscriber>>
+  ): void;
+
+  setRequestValidator(topic: string, validator: RequestValidator): void;
```

Comment:
Counter proposal:

How would you feel about

```ts
addHttpPostIngressEndpoint(options: {
  topic: string,
  validator?: RequestValidator | undefined,
  // ... other properties in the future
});
```

This keeps it ALL in code, no need to go to the config to set this up. Instead, the modules get to decide. For example, now you can do

```ts
.addModule(githubWebhookIngressModule({ topic: 'github' }))
.addModule(githubWebhookIngressModule({ topic: 'githubEnterprise', validator: doOurCustomValidation })
```

or whatnot, and the routes "just magically appear".

Reply:
(the naming is totally open to suggestions here btw)

Reply:
In general, I prefer a no-code option, however I thought about a programatic way as well when playing with registering the validator at my other branch.

```
  .add(githubWebhookIngressEventsModule({
     topic: 'githubEnterprise',
     validator: doOurCustomValidation,
  })
```
I would even do that inside of the module itself.

I will ad a change to cover that.

Reply:
more or less, I went for your interface suggestion. I think, overall it brought some more improvements.